caravel/models.py

```diff
@@ -1973,10 +1982,74 @@ def to_dict(self):
             'tempTable': self.tmp_table_name,
             'userId': self.user_id,
         }
+
     @property
     def name(self):
         ts = datetime.now().isoformat()
         ts = ts.replace('-', '').replace(':', '').split('.')[0]
         tab = self.tab_name.replace(' ', '_').lower() if self.tab_name else 'notab'
         tab = re.sub(r'\W+', '', tab)
         return "sqllab_{tab}_{ts}".format(**locals())
+
+
+ROLES_BLACKLIST = set(['Admin', 'Alpha', 'Gamma', 'Public'])
+
+
+class DatasourceAccessRequest(Model, AuditMixinNullable):
+    """ORM model for the access requests for datasources and dbs."""
+    __tablename__ = 'access_request'
+    id = Column(Integer, primary_key=True)
+
+    table_id = Column(Integer, ForeignKey('tables.id'))
+    druid_datasource_id = Column(Integer, ForeignKey('datasources.id'))
+
+    table = relationship('SqlaTable', foreign_keys=[table_id])
+    druid_datasource = relationship(
+        'DruidDatasource', foreign_keys=[druid_datasource_id])
+
+    @property
+    def username(self):
+        return self.creator()
+
+    @property
+    def datasource(self):
+        if self.druid_datasource:
+            return self.druid_datasource
+        return self.table
+
+    @property
+    def datasource_url(self):
```

Comment:
Rename to `datasource_link` 
I'd except a link to be a `<a>` and a url to be `http://...`